lib/sequence.js

```diff
@@ -167,6 +167,40 @@ Sequence.prototype._setHooks = function() {
             done();
         }.bind(doc));
     });
+
+    this._schema.pre('findOneAndUpdate', function(next) {
+        // jscs:disable
+        var _thisquery = this,
+            referenceValue;
+
+        // jscs:enable
+
+        this._update['$setOnInsert'] = this._update['$setOnInsert'] || {};
+
+        referenceValue = _this._getCounterReferenceField(this._update['$setOnInsert']);
+        _this._setNextCounter(_this._options.reference_fields, referenceValue, function(err, seq) {
```

Comment:
Hmm, I'll have to see if there is a parameter that can be used to check for that scenario.
Otherwise, we'd have to do a db query to look for the object id...